mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/TaskExecutorRegistration.java

```diff
@@ -141,4 +147,29 @@ public Optional<String> getAttributeByKey(String attributeKey) {
 
         return Optional.empty();
     }
+
+    @JsonIgnore
+    public Map<String, String> getSchedulingAttributes() {
+        return taskExecutorAttributes.entrySet().stream()
+            .flatMap(entry -> {
+                Matcher matcher = WorkerConstants.MANTIS_SCHEDULING_ATTRIBUTE_PATTERN.matcher(entry.getKey());
+                return matcher.matches() ? Stream.of(new AbstractMap.SimpleEntry<>(matcher.group(1).toLowerCase(), entry.getValue())) : Stream.empty();
+            })
+            .collect(Collectors.toMap(
+                Map.Entry::getKey,
+                entry -> getAttributeByKey(entry.getKey()).orElse(entry.getValue())
+            ));
+    }
+
+    @JsonIgnore
+    public TaskExecutorGroupKey getTaskExecutorGroupKey() {
+        return TaskExecutorGroupKey.of(machineDefinition, getSchedulingAttributes());
+    }
+
+    @Value
+    @AllArgsConstructor(staticName = "of")
```

Comment:
nit: just curious, what's the need for this static constructor here?

Reply:
we don't anymore